### PR TITLE
DAT-3024

### PR DESCRIPTION
**Pull request by dianafa**
https://wikia-inc.atlassian.net/browse/DAT-3024

This PR introduces a new component responsible for the main infobox image; the InfoboxImageMediaComponent.

It is responsible for computing computedHeight and url which were previously using ugly "is infobox image" checks in the ImageMediaComponent.  Correctly sets also the `caption` and `limitHeight` properties.

@rogatty

### CODE REVIEW

File: front/scripts/main/components/InfoboxImageMediaComponent.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
+/// <reference path="../app.ts" />
+/// <reference path="./ImageMediaComponent.ts" />
+/// <reference path="../mixins/ViewportMixin.ts" />
+'use strict';
+
+App.InfoboxImageMediaComponent = App.ImageMediaComponent.extend(App.ViewportMixin, {
```
Comment (RafLeszczynski) [inline front/scripts/main/components/InfoboxImageMediaComponent.ts]: I wonder if ti wouldn't be better to name this `InfoboxHeroImageMediaComponent` cause this what it's all about, not all images inside infobox?

**Comment by RafLeszczynski** [comment]
Just one small naming convention question and LGTM

File: front/scripts/main/components/InfoboxImageMediaComponent.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	 * Extended version of ImageMediaComponent#computedHeight.
+	 * Takes into account cropping main infobox images and basing on it's dimensions sets cropping mode.
+	 * @return number
+	 */
+	computedHeight: Em.computed('viewportDimensions.width', 'media.width', 'media.height', function (): number {
+		var windowWidth: number = this.get('viewportDimensions.width'),
+			imageAspectRatio: number = this.get('imageAspectRatio'),
+			imageWidth: number = this.get('media.width') || windowWidth,
+			imageHeight: number = this.get('media.height'),
+			maximalWidth: number = Math.floor(imageHeight * imageAspectRatio),
```
Comment (rogatty) [inline front/scripts/main/components/InfoboxImageMediaComponent.ts]: How about `maxWidth` like everywhere else?

File: front/scripts/main/components/InfoboxImageMediaComponent.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
+/// <reference path="../app.ts" />
+/// <reference path="./ImageMediaComponent.ts" />
+/// <reference path="../mixins/ViewportMixin.ts" />
+'use strict';
+
+App.InfoboxImageMediaComponent = App.ImageMediaComponent.extend(App.ViewportMixin, {
+	imageAspectRatio: 16 / 9,
+	caption: false,
+	limitHeight: true,
+	noNormalizeWidth: true,
```
Comment (rogatty) [inline front/scripts/main/components/InfoboxImageMediaComponent.ts]: This is strange name. Maybe `normalizeWidth`? Also set the default in `MediaComponent` explicitly.

File: front/scripts/main/components/InfoboxImageMediaComponent.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+		return computedHeight;
+	}),
+
+	/**
+	 * @desc return the params for getThumbURL for infobox image.
+	 * In case of very high or very wide images, crop them properly.
+	 */
+	url: Em.computed('media', 'computedHeight', 'imageSrc', {
+		get(): string {
+				var media: ArticleMedia = this.get('media'),
```
Comment (rogatty) [inline front/scripts/main/components/InfoboxImageMediaComponent.ts]: Too many `\t`

File: front/scripts/main/components/MediaComponent.ts [hunk 5]
```diff
@@ -93,6 +90,10 @@ App.MediaComponent = Em.Component.extend(App.VisibleMixin, {
 
 App.MediaComponent.reopenClass({
 	newFromMedia: function (media: ArticleMedia): typeof App.MediaComponent {
+		if (media.context === 'infobox') {
```
Comment (rogatty) [inline front/scripts/main/components/MediaComponent.ts]: Let's put it in the chain below. Maybe as the first `else if`?

**Comment by rogatty** [comment]
Fix all the small things and LGTM.

**Pull request merged**